### ISSUE TITLE
[feat] `no-set-state-directly-in-use-effect`

### ISSUE THREAD

**Issue by SukkaW** [Rel1cx/eslint-react#628]
### Describe the problem

Common bad practice:

```tsx
const Comp = () => {
  const [data, setData] = useState();
  useEffect(() => {
    setData();
  }, []);
}
```

### Describe the solution you'd like

Create a rule `no-set-state-directly-in-use-effect` to prevent set state directly in the top useEffect callback scope:

**Invalid**

```tsx
const Comp = () => {
  const [data, setData] = useState();
  useEffect(() => {
    setData();
  }, []);
}
```

**Valid**

```tsx
const Comp = () => {
  const [data, setData] = useState();
  useEffect(() => {
   fetch().then(() => setData());
  }, []);
}
```

```tsx
const Comp = () => {
  const [data, setData] = useState();
  useEffect(() => {
   (async () => { setData() })();
  }, []);
}
```

```tsx
const Comp = () => {
  const [data, setData] = useState();
  useEffect(() => {
   const onLoad = () => {
     setData();
   };
  }, []);
}
```
